feature_layers/create-mobile-geodatabase/src/main/java/com/esri/samples/create_mobile_geodatabase/CreateMobileGeodatabaseSample.java

```diff
@@ -0,0 +1,423 @@
+          if (!tableStage.isShowing()){
+            showTableWindow = false;
+          }
+        }
+
+        if(showTableWindow && tableStage != null) {
+          // close table window if open
+          tableStage.close();
+          showTableWindow = false;
+        }
+        else {
+          // create and show feature table in a new window
+          displayTable(stage);
+        }
+      });
+
+      // add the map view and UI elements to the stack pane
+      stackPane.getChildren().addAll(mapView, vBoxControls);
+      StackPane.setAlignment(vBoxControls, Pos.TOP_LEFT);
+      StackPane.setMargin(vBoxControls, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new class as data model to hold feature data.
+   * Instances of this class are used to populate the TableView.
+   */
+  @SuppressWarnings("unused")
+  public static class GeoFeature{
+    private final SimpleStringProperty oid;
+    private final SimpleStringProperty timestamp;
+
+    private GeoFeature(String oid, String timestamp) {
+      this.oid = new SimpleStringProperty(oid);
+      this.timestamp = new SimpleStringProperty(timestamp);
+    }
+
+    public String getOid() {
+      return oid.get();
+    }
+
+    public String getTimestamp() {
+      return timestamp.get();
+    }
+  }
+
+  /**
+   * Creates UI with two buttons and a label component.
+   */
+  private void setUpControlsVBox() {
+
+    // create label to show number of added features
+    label = new Label("Number of features added: ");
+    label.setStyle("-fx-text-fill: white");
+
+    // create button to show alert and close geodatabase
+    createGeodatabaseButton = new Button("Create Mobile Geodatabase");
+    createGeodatabaseButton.setAlignment(Pos.CENTER);
+    createGeodatabaseButton.setDisable(true);
+
+    // create button to show features stored in the geodatabase
+    viewTableButton = new Button("View Table");
+    viewTableButton.setAlignment(Pos.CENTER);
+    viewTableButton.setDisable(true);
+
+    // create and configure a VBox
+    vBoxControls = new VBox(10);
+    vBoxControls.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.7)"),
+      CornerRadii.EMPTY, Insets.EMPTY)));
+    vBoxControls.setPadding(new Insets(10.0));
+    vBoxControls.setMaxSize(185, 100);
+    vBoxControls.setDisable(false);
+
+    // add the label and button to the VBos
+    vBoxControls.getChildren().addAll(label, createGeodatabaseButton, viewTableButton);
+  }
+
+  /**
+   * Create and load a new geodatabase file and feature layer from table description fields
+   */
+  private void createGeodatabase() throws IOException {
+
+    // create a directory for the geodatabase file
+    Path directoryPath = Paths.get(System.getProperty("user.dir"), "CreateMobileGeodatabase");
+    if(!Files.exists(directoryPath))
+      Files.createDirectories(directoryPath);
```

Comment:
for readability, best Java practice is to keep the braces.

```suggestion
    // if the directory path doesn't already exist, create it
    if(!Files.exists(directoryPath)) {
      Files.createDirectories(directoryPath);
      }
```